http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/sam/SamAuthProvider.scala

```diff
@@ -211,7 +226,7 @@ class SamAuthProvider[F[_]: Effect: Logger](samDao: SamDAO[F],
 
   private def hasRuntimePermissionInternal(
     samResource: RuntimeSamResource,
-    action: LeoAuthAction,
```

Comment:
why did we introduce LeoAuthAction?

Reply:
This is part of the changes Azmina made in her PR. 

LeoAuthAction gets extended by RuntimeAction/ProjectAction/PersistentDiskAction

Reply:
Yeah.. makes sense that these methods take specific types

Reply:
I might try to remove `LeoAuthAction` in a future PR...It makes inheritance more complicated and doesn't bring much value